src/lib/queue.js

```diff
@@ -30,7 +34,8 @@ export default function(opts) {
 
         const state = (error ? 'FAIL' : 'SUCCESS');
 
-        const newState = {state, error};
+        // remove circular references from error, and parse it back to an object.
+        const newState = {state, error: JSON.parse(stringifySafe(error))};
```

Comment:
1. move `JSON.parse(stringifySafe(error))` into function with sensible name
* re-use in next `catch`-block too